lib/ansible/modules/windows/win_product_facts.ps1

```diff
@@ -57,9 +57,59 @@ if (-not $product_key) {
     }
 }
 
+try {
+    $license_info = Get-CimInstance SoftwareLicensingProduct | Where-Object PartialProductKey
+
+    switch ($license_info.LicenseStatus){
+        0 {
+            $winlicense_status="Unlicensed"
+        }
+
+        1 {
+            $winlicense_status="Licensed"
+        }
+
+        2 {
+            $winlicense_status="OOBGrace"
+        }
+
+        3 {
+            $winlicense_status="OOTGrace"
+        }
+
+        4 {
+            $winlicense_status="NonGenuineGrace"
+        }
+
+        5 {
+            $winlicense_status="Notification"
+        }
+
+        6 {
+            $winlicense_status="ExtendedGrace"
+        }
+
+        default {
+            $winlicense_status="NA"
+        }
+    }
+
+    $winlicense_edition = $license_info.Name
+    $winlicense_channel = $license_info.ProductKeyChannel
+
+} catch {
+    $winlicense_edition = "NA"
```

Comment:
Same comments for these vars around NA vs N/A.

Reply:
null ;-)